File: designpattern/src/main/java/com/seniorlibs/designpattern/ch25v2/viewer/StatViewer.java
package com.seniorlibs.designpattern.ch25v2.viewer;

import com.seniorlibs.designpattern.ch25v2.model.RequestStat;

import java.util.Map;

public interface StatViewer {
    void output(Map<String, RequestStat> requestStats, long startTimeInMillis, long endTimeInMills);
}

File: designpattern/src/main/java/com/seniorlibs/designpattern/ch25v3/viewer/StatViewer.kt
package com.seniorlibs.designpattern.ch25v3.viewer

import com.seniorlibs.designpattern.ch25v3.model.RequestStat

interface StatViewer {
    fun output(
        requestStats: Map<String, RequestStat>,
        startTimeInMillis: Long,
        endTimeInMills: Long
    )
}